### PR TITLE
Add a context manager API for gradient caching in DVEmbAttributor

### PR DESCRIPTION
**Pull request by VanderLFZ**
This PR implements what is proposed in #249

### CODE REVIEW

File: dattri/algorithm/dvemb.py [hunk 1]
```diff
@@ -274,10 +275,74 @@ def cache_gradients(
...
         else:
             self._cache_full_gradients(epoch, batch_data, indices, learning_rate)
 
+    @contextmanager
+    def cache_gradients_context(
+        self,
+        epoch: int,
+        indices: Tensor,
+        learning_rate: float,
+    ):
```
Comment (TheaperDeng) [inline dattri/algorithm/dvemb.py]: We need a docstring for this context manager, explaining what does each paramter mean.

**Comment by VanderLFZ** [comment]
I’ve handled all of them. Let me know if anything else needs attention.

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Comment by TheaperDeng** [comment]
Please fix the docstring and coding style issues as stated in github actions.
